### ISSUE TITLE
Can't run on termux due to old node.js version

### ISSUE THREAD

**Issue by danielkrajnik** [200ok-ch/organice#962]
**Describe the bug**
package.json requires version 12. This is a problem, because there is no way to install this version on termux. Binary distribution is no longer supported: `curl -fsSL https://deb.nodesource.com/setup_12.x | bash - && pkg install nodejs` won't work and because termux uses $PREFIX same as nvm, which makes it unavailable.

If there is another way to run organice on android please let me know.

**To Reproduce**
1. git clone https://github.com/200ok-ch/organice
3. cd organice
4. yarn install
5. See error

**Expected behavior**
Use nodejs 20.2.0

**Desktop (please complete the following information):**
 - OS: Android

**Smartphone (please complete the following information):**
 - Device: google Pixel 6
 - OS: GrapheneOS

**Comment by danielkrajnik** [comment]
I've tried just changing version in package.json to 20.2.0, but got an error when running yarn install:
`error /data/data/com.termux/files/home/organice/node_modules/node-sass: Command failed.`

**Comment by danielkrajnik** [comment]
by the way version 12 [isn't supported](https://nodejs.dev/en/about/releases/) anymore?

**Comment by branch14** [comment]
Hi @danielkrajnik, first of all let me express my admiration of your bravery. :)

Secondly, May I ask what your use case is for running organice within Termux?

Organcie is a PWA with offline capabilities which syncs to a storage backend of your choosing when online. If you just want to use organice on your phone pointing a browser to any instance e.g. https://organice.200ok.ch should be enough.

If you have a use case for hosting a PWA within Termux on an Android device consider me intrigued. I'd love to hear about it.

**Comment by munen** [comment]
Hi @branch14 

thanks for chiming in. Op wants to access local files. There’s prior art and documentation on how to run organice in Termux with a local WebDAV server to accomplish that: https://github.com/200ok-ch/organice/tree/master/contrib

**Comment by branch14** [comment]
IC. Nevermind my patronizing scrutiny then. :)

**Comment by danielkrajnik** [comment]
Thanks both @munen [the guide](https://github.com/200ok-ch/organice/tree/master/contrib/organice-on-android-with-webdav-server) you mentioned is outdated though - termux repositories [only offer the latest version of node](https://wiki.termux.com/index.php?title=Node.js) and as of june 2023 it's node 20.2.0 - this won't work. nvm isn't available neither compiling from source.
@branch14 I'm trying to self host the app and use local files. The idea was to serve them as locally with a webdav server ([dufs](https://github.com/sigoden/dufs)). I think it's important to be able to self-host if you care about privacy.

**Comment by munen** [comment]
@danielkrajnik 

> so I wonder if given that organice is a client-side SPA wouldn't it be possible to just bundle it with something like webpack and serve its html, css and js with another http server (on localhost)? That would be much easier.

Yes, you can just compile organice on any machine and host it on any kind of webserver. That's exactly how we host organice on https://organice.200ok.ch, in fact(;

 Here's the doc for that: https://organice.200ok.ch/documentation.html#deployment

If you have further questions, feel free to re-open this issue!
